### ISSUE TITLE
Make HttpAsyncClientBuilder implementation configurable

### ISSUE THREAD

**Issue by gquintana** [elastic/elasticsearch#58428]
The `HttpAsyncClientBuilder` implementation creation is hard coded in 
https://github.com/elastic/elasticsearch/blob/v7.8.0/client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java#L209

It prevents using custom implementations like Zipkin one:
https://github.com/openzipkin/brave/issues/1226

I would like to be able to replace
```
HttpAsyncClientBuilder.create()
```
with
```
TracingHttpAsyncClientBuilder.create(tracing)
```

**Comment by jbaiera** [comment]
Is this builder something you could create and inject back to the elasticsearch client via the configuration [callback interface](https://github.com/elastic/elasticsearch/blob/v7.8.0/client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java#L215)? It would require mirroring the defaults from the client (ehhhhhh not great), and potentially wrapping any third-party user callbacks, but it could get you unblocked in the mean time?

**Comment by gquintana** [comment]
Yes, copying code from `RestClientBuilder` is a workaround, but not very satisfactory.
Each time we upgrade Elasticsearch client library we must think about resynchronizing code.
A better solution would be to introduce an`HttpAsyncClientBuilder` factory which would default to `HttpAsyncClientBuilder::create` and could be changed.

**Comment by nilu-source** [comment]
Any update on this PR?
